lib/util/exec/index.ts

```diff
@@ -22,6 +23,8 @@ export interface ExecOptions extends ChildProcessExecOptions {
   extraEnv?: Opt<ExtraEnv>;
   docker?: Opt<DockerOptions>;
   toolConstraints?: Opt<ToolConstraint[]>;
+  preCommands?: DockerExtraCommands;
```

Comment:
We should move all interfaces to `./types.ts`

Reply:
I'm going to be refactoring these all quite a lot, so will do it later